### ISSUE TITLE
Test failures with boost 1.81

### ISSUE THREAD

**Issue by loqs** [BRiAl/BRiAl#53]
BRiAl 1.21.11
boost 1.81.0
gcc 12.2.0

```
./configure --prefix=/usr --enable-shared
make
make check
```
tests/test-suite.log
```
========================================
   BRiAl 1.2.11: tests/test-suite.log
========================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: unittests
===============

Running 122 test cases...
BoolePolynomialTest.cc(352): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(1, ring).stableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(353): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x).stableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(354): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x*y).stableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(355): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x*y + z).stableHash() == 15553547522801749128u has failed [15080868938440987258 != 15553547522801749128][0;39;49m
BoolePolynomialTest.cc(356): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x*y + z + 1).stableHash() == 14746308500923705671u has failed [4046934837422409843 != 14746308500923705671][0;39;49m
BoolePolynomialTest.cc(357): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x*y + x + z).stableHash() == 6609543853578108291 has failed [17287305169241640544 != 6609543853578108291][0;39;49m
BoolePolynomialTest.cc(358): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_independent": check poly_type(x*y + x + z + 1).stableHash() == 13745298887294022704u has failed [15334652091369486486 != 13745298887294022704][0;39;49m
BoolePolynomialTest.cc(605): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly_type(1, ring).leadStableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(606): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly_type(a).leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(607): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly_type(a*b).leadStableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(608): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly_type(c*d +c + e + 1).leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(610): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly1.leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(611): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_lp": check poly2.leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(725): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly_type(1, ring).leadStableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(726): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly_type(a).leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(727): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly_type(a*b).leadStableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(728): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly_type(c*d +c + e + 1).leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(730): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly1.leadStableHash() == 13332077148710286618u has failed [14792830888691653104 != 13332077148710286618][0;39;49m
BoolePolynomialTest.cc(731): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dlex": check poly2.leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(838): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly_type(1, ring).leadStableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(839): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly_type(a).leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(840): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly_type(a*b).leadStableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(841): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly_type(c*d +c + e + 1).leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(843): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly1.leadStableHash() == 16171595639562729641u has failed [16452769785566003032 != 16171595639562729641][0;39;49m
BoolePolynomialTest.cc(844): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_dp_asc": check poly2.leadStableHash() == 298249910754104916 has failed [7788958613296501621 != 298249910754104916][0;39;49m
BoolePolynomialTest.cc(953): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly_type(1,ring).leadStableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(954): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly_type(a).leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(955): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly_type(a*b).leadStableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(956): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly_type(c*d +c + e + 1).leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(958): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly1.leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(959): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dlex": check poly2.leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(1068): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly_type(1, ring).leadStableHash() == 16982317933699389295u has failed [6643957975231536490 != 16982317933699389295][0;39;49m
BoolePolynomialTest.cc(1069): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly_type(a).leadStableHash() == 14511566905035657723u has failed [14305226453458947074 != 14511566905035657723][0;39;49m
BoolePolynomialTest.cc(1070): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly_type(a*b).leadStableHash() == 1160268684422367443 has failed [7488647634487918362 != 1160268684422367443][0;39;49m
BoolePolynomialTest.cc(1071): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly_type(c*d +c + e + 1).leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m
BoolePolynomialTest.cc(1073): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly1.leadStableHash() == 16171595639562729641u has failed [16452769785566003032 != 16171595639562729641][0;39;49m
BoolePolynomialTest.cc(1074): [1;31;49merror: in "BoolePolynomialTestSuite/test_ordering_block_dp_asc": check poly2.leadStableHash() == 994909422134781907 has failed [8683732623732266773 != 994909422134781907][0;39;49m

[1;31;49m*** 37 failures are detected in the test module "UnitTests"
[0;39;49mFAIL unittests (exit status: 201)

```

**Comment by kiwifb** [comment]
Thank you for the report. It looks like hashes have changed again. Annoying but not that important. I'll get a fix in in the next couple of days and cut a release.

**Comment by kiwifb** [comment]
A  big issue is that those hash value depends somewhat of the version of boost and even possibly the compiler. Testing for value in that way is not very future proof. I may just remove those particular tests.

**Comment by kiwifb** [comment]
Should be fixed in 1.2.12. Re-open if you still have problems.
